common4j/src/main/com/microsoft/identity/common/java/controllers/CommandDispatcher.java

```diff
@@ -641,7 +628,17 @@ public static void beginInteractive ( final InteractiveTokenCommand command){
                 //Cancel interactive request if authorizationInCurrentTask() returns true OR this is a broker request.
                 if (LibraryConfiguration.getInstance().isAuthorizationInCurrentTask() || command.getParameters() instanceof BrokerInteractiveTokenCommandParameters) {
                     // Send a broadcast to cancel if any active auth request is present.
-                    LocalBroadcaster.INSTANCE.broadcast(CANCEL_AUTHORIZATION_REQUEST, new PropertyBag());
+                    if(LocalBroadcaster.INSTANCE.hasReceivers(CANCEL_AUTHORIZATION_REQUEST)) {
+                        LocalBroadcaster.INSTANCE.broadcast(CANCEL_AUTHORIZATION_REQUEST, new PropertyBag());
+                    } else if (LocalBroadcaster.INSTANCE.hasReceivers(RETURN_AUTHORIZATION_REQUEST_RESULT)) {
+                        // This means that there is previous interactive request(s) waiting for Authorization result, however
+                        // the actual authorization process for that request has not been started as there are no registered listener/receivers
+                        // for "CANCEL_AUTHORIZATION_REQUEST". This results in the sInteractiveExecutor thread being in deadlock
+                        // state and not able to process any more interactive requests. To get out of this deadlock and be able to
+                        // process future interactive request we need to shutdown and restart the sInteractiveExecutor executor service.
+                        sInteractiveExecutor.shutdownNow();
```

Comment:
We probably want to make sure that callbacks of all the pending tasks are executed.

1. Make sure that exception from Thread.interrupt() are handled and returned to MSAL properly (as SDK_AUTH_CANCEL?)
2. For those that are not canceled properly... (javadoc of this method says it would return those runnables as a list)... we might want to invoke the callbacks manually. Should we create a new Runnable type that takes in / contains the command object?

Reply:
1. Added error messaging for InterruptedException.
2. This is a corner case and we should rarely run into this (that too only one thread), IMO returning an error/exception should be fine for now.